runtime/src/accounts_db.rs

```diff
@@ -560,25 +560,77 @@ impl AccountsDB {
         false
     }
 
+    // Purge zero lamport accounts for garbage collection purposes
+    // Only remove those accounts where the entire rooted history of the account
+    // can be purged because there are no live append vecs in the ancestors
     pub fn purge_zero_lamport_accounts(&self, ancestors: &HashMap<u64, usize>) {
         self.report_store_stats();
+        let mut purges = HashMap::new();
```

Comment:
Yeah, this is exactly what I thought would have liked to see! But not bothered to write a comment. :p This way, we can reduce redundant call of `would_purge` around [here](https://github.com/solana-labs/solana/pull/7157/files#diff-2099c5256db4eb5975c8834af38f6456R598). :)